modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/RemoveConstantsFromQueryTest.java

```diff
@@ -0,0 +1,179 @@
+import org.apache.ignite.internal.processors.cache.query.GridCacheQueryType;
+import org.apache.ignite.internal.processors.performancestatistics.AbstractPerformanceStatisticsTest;
+import org.apache.ignite.internal.util.typedef.F;
+import org.apache.ignite.lang.IgniteBiTuple;
+import org.apache.ignite.testframework.junits.WithSystemProperty;
+import org.junit.Test;
+
+import static org.apache.ignite.IgniteSystemProperties.IGNITE_TO_STRING_INCLUDE_SENSITIVE;
+import static org.apache.ignite.internal.processors.performancestatistics.AbstractPerformanceStatisticsTest.cleanPerformanceStatisticsDir;
+import static org.apache.ignite.internal.processors.performancestatistics.AbstractPerformanceStatisticsTest.startCollectStatistics;
+import static org.apache.ignite.internal.processors.performancestatistics.AbstractPerformanceStatisticsTest.stopCollectStatisticsAndRead;
+
+/**
+ * Tests check that with {@link IgniteSystemProperties#IGNITE_TO_STRING_INCLUDE_SENSITIVE} == false literals from query
+ * will be deleted from query before logging it to history, events, profiling tool.
+ */
+public class RemoveConstantsFromQueryTest extends AbstractIndexingCommonTest {
+    /** {@inheritDoc} */
+    @Override protected IgniteConfiguration getConfiguration(String igniteInstanceName) throws Exception {
+        return super.getConfiguration(igniteInstanceName).setIncludeEventTypes(EventType.EVT_SQL_QUERY_EXECUTION);
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void beforeTestsStarted() throws Exception {
+        cleanPerformanceStatisticsDir();
+    }
+
+    /**  */
+    @Test
+    @WithSystemProperty(key = IGNITE_TO_STRING_INCLUDE_SENSITIVE, value = "false")
+    public void testConstantRemoved() throws Exception {
+        IgniteEx ignite = startGrid(0);
+
+        ignite.cluster().active(true);
+
+        cleanPerformanceStatisticsDir();
+
+        startCollectStatistics();
+
+        List<IgniteBiTuple<String, String>> qries = Arrays.asList(
+            F.t("CREATE TABLE TST(id INTEGER PRIMARY KEY, name VARCHAR, age integer)", null),
+
+            F.t("CREATE TABLE TST2(id INTEGER PRIMARY KEY, name VARCHAR, age integer)", null),
+
+            F.t("INSERT INTO TST(id, name, age) VALUES(1, 'John Connor', 16)",
+                "INSERT INTO PUBLIC.TST( ID, NAME, AGE ) VALUES (?, ?, ?)"),
+
+            F.t("INSERT INTO TST SELECT id, name, age FROM TST2 WHERE name = 'John Connor'",
+                "INSERT INTO PUBLIC.TST( ID, NAME, AGE )  SELECT ID, NAME, AGE FROM PUBLIC.TST2 WHERE NAME = ?"),
+
+            F.t("UPDATE TST SET name = 'Sarah Connor' WHERE id = 1",
+                "UPDATE PUBLIC.TST SET NAME = ? WHERE ID = ?"),
+
+            F.t("DELETE FROM TST WHERE name = 'Sarah Connor'",
+                "DELETE FROM PUBLIC.TST WHERE NAME = ?"),
+
+            F.t("SELECT * FROM TST WHERE name = 'Sarah Connor'",
+                "SELECT __Z0.ID, __Z0.NAME, __Z0.AGE FROM PUBLIC.TST __Z0 WHERE __Z0.NAME = ?"),
```

Comment:
Here we rely on the underlying H2 version. In case of changes in H2 or moving SQL engine to Calcite generated SQL can be different. Perhaps it's better to just check query only for the absence of restricted constant, but not for equality.

Reply:
I think we can just change this test when we will migrate to another SQL engine.

Reply:
I think it's fragile logic to assert values not acknowledged by any contract received from the external source. `__Z0` appears in this query not because any user expects it here, but because we have checked it against the black box and got such results. At least checking for some reasonable pattern will be better here (`SELECT.*FROM.*TST.*WHERE.*NAME.*=.*\?` or something like this)

Reply:
OK. I reworked the test to use regexp.